### PR TITLE
Add --keep-mock option

### LINKED ISSUE #314: some way to skip replacing mock with unittest.mock
when you're using python 3.6 or 3.7 `unittest.mock.AsyncMock` is unavailable, but `mock.AsyncMock` is available if you're using the latest backport

there should be some way to skip replacing `mock` with `unittest.mock`, or the fixer should be moved to `--py38-plus` and then to a later `--py...-plus` as and when `unittest.mock` gains features

### PR DESCRIPTION
**Pull request by mxr**
Resolves #314

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -243,6 +243,7 @@ Availability:
 
 Availability:
 - `--py3-plus` is passed on the commandline.
+- Can be suppressed with `--no-mock`. See [#314](https://github.com/asottile/pyupgrade/issues/314) for more context.
```
Comment (asottile) [inline README.md:246]: I'd just say `- [Unless `--no-mock` is passed on the commandline`]()`

File: pyupgrade.py [hunk 2]
```diff
@@ -1994,7 +1994,9 @@ def _replace_yield(tokens: List[Token], i: int) -> None:
     tokens[i:block.end] = [Token('CODE', f'yield from {container}\n')]
 
 
-def _fix_py3_plus(contents_text: str, min_version: MinVersion) -> str:
+def _fix_py3_plus(
+    contents_text: str, min_version: MinVersion, no_mock: bool = False,
```
Comment (asottile) [inline pyupgrade.py:1998]:
```suggestion
        contents_text: str,
        min_version: MinVersion,
        no_mock: bool = False,
```

File: pyupgrade.py [hunk 3]
```diff
@@ -2198,10 +2200,10 @@ def _replace(i: int, mapping: Dict[str, str], node: NameOrAttr) -> None:
         elif token.offset in visitor.io_open_calls:
             j = _find_open_paren(tokens, i)
             tokens[i:j] = [token._replace(name='NAME', src='open')]
-        elif token.offset in visitor.mock_mock:
+        elif token.offset in visitor.mock_mock and not no_mock:
```
Comment (asottile) [inline pyupgrade.py:2203]: better would be to pass this into the Visitor -- that way we can avoid the expensive tokenization if all that was found to "fix" was mock stuff

File: tests/mock_test.py [hunk 4]
```diff
@@ -4,20 +4,30 @@
...
+            ((3,), True),
+            id='does not rewrite if no_mock==True',
+        ),
+
     ),
 )
-def test_mock_noop(s):
-    assert _fix_py3_plus(s, (3,)) == s
+def test_mock_noop(s, args):
+    assert _fix_py3_plus(s, *args) == s
```
Comment (asottile) [inline tests/mock_test.py:30]: I would make another test probably, this is trying too hard to abuse parametrize imo

Comment (mxr) [reply tests/mock_test.py:30]: Sure. FWIW my fork of pytest doesn't consider this abuse

File: pyupgrade.py (hunk 3 above)
Comment (mxr) [reply pyupgrade.py:2203]: Sure

File: pyupgrade.py (hunk 2 above)
Comment (mxr) [reply pyupgrade.py:1998]: > any color you like

File: README.md (hunk 1 above)
Comment (mxr) [reply README.md:246]: Sure, should I provide more context on why we have this flag? Either by linking to the PR or summarizing something like "`AsyncMock` shouldn't be rewritten in certain versions and for consistent behavior across files you can disable rewriting mock entirely"

Comment (asottile) [reply README.md:246]: oh, my template was supposed to look like a link but I forgot to paste the link into it -- I'd just link to the issue

File: tests/mock_test.py (hunk 4 above)
Comment (asottile) [reply tests/mock_test.py:30]: your, _more serious_ fork?

Comment (mxr) [reply tests/mock_test.py:30]: Yes I've been up all night redeyed working on it

Comment (mxr) [reply tests/mock_test.py:30]: Based on your comments I can see you're greeneyed with envy

**Comment by mxr** [comment]
I'll aim to finish this by tomorrow 🤞

**Comment by mxr** [comment]
(But if there's a pressing need to get this out feel free to take over this PR)

**Pull request merged**